### DISCUSSION TITLE
Defining an orthotropic material based on the Hill model

### DISCUSSION THREAD

**Discussion by ghost** [thelfer/tfel#324] [Q&A] [answered]
Hi there,

our team is creating a material model for cardboard, which is an orthothropic material. Our goal is to simulate the compaction behaviour of cardboard, starting from the Hill model.

We're currently moving development to MFront (and simulating in FEniCS) for performance reasons and accessibility of the results. Unfortunately, while translating our material model into MFront we've run into a few difficulties & general questions, so we hope that you can give us a few hints. 

We've started out by using the predefined example showing how to implement an orthotropic material using the Hill Tensor (https://tfel.sourceforge.net/orthotropiclinearhardeningplasticity.html).

However, when running our simulation on a simple UnitCube, we are getting unexpected results. When pressing down on the UnitCube ("compacting"), it shrinks in Z direction and expands in Y direction as expected, but also unexpectedly shrinks in the X direction.

We suspect this might be because in the provided example, in the `@InitLocalVariables` and `@Integrator` block, there is a Hill equivalent stress (`seq`) defined, although for the Hill model itself, such a value doesn't really exist. We also checked the book of Lemaître et Chaboche, but could not find a definition of this equivalent stress. 

From our understanding, using the value obtained by the Hill formula, one can only determine if the material does yield (boolean `b`). So the given example might only work for cases where the resulting Hill coefficients (calculated from the ratios of the yield stresses) do not differ from each other significantly. But this is not the case for cardboard, where the yield stresses differ a lot between the axes. This results in a negative Hill coefficient (line 25 in the MFront file), hence the shrinking in X direction.

If this is indeed the case, we'd need to alter the `@Integrator` block to represent that there actually isn't an equivalent stress `seq` in the Hill model. But we are at a loss on how exactly we would replace the `seq` and `iseq` variables for the elastic and plastic update (calculating `feel`, `dfeel_ddeel` etc.), since the implementation we're moving on from used a different logic. So we'd be grateful for any hints and / or tips.

I've attached the Hill .mfront file (`Hilloriginal.mfront`) with our cardboard parameters and our FEniCS `UnitCube.py` file for the simulation. If you can't run the script, I have also attached the resulting file `displacement_z.png` showing the shrinking in X direction.

Many thanks in advance and best regards,

Ruben Heinrich

[Hill_files.zip](https://github.com/thelfer/tfel/files/9838834/Hill_files.zip)

**Comment by thelfer** [comment, marked as answer]
Hi @aheinri5

Thanks for your interest. 

Indeed, the traction curve seems quite strange

![strain](https://user-images.githubusercontent.com/13975528/197246464-39b682fb-d02f-4d61-9f84-b42853b17e0e.png)

Here are a simple test with `MTest`: [test.zip](https://github.com/thelfer/tfel/files/9840894/test.zip)

The definition of the Hill equivalent stress used in MFront is defined here: <https://thelfer.github.io/tfel/web/tensors.html>

This definition clearly shows that the Hill coefficients can't be negative. 

However, there are a few statements that I don't get in your text:

> here is a Hill equivalent stress (seq) defined, although for the Hill model itself, such a value doesn't really exist.

The model described here is classical and well defined. Elasticity is given by the orhotropic Hooke law and the yield surface is defined by: 

seq -R(p) = 0

So I don't get what you mean by  "such a value doesn't really exist." Could you elaborate ?

> If this is indeed the case, we'd need to alter the @Integrator block to represent that there actually isn't an equivalent stress seq in the Hill model. 

If you don't mind, could you write the consistutive equations of what you call the Hill model. I think that we are not talking about the same thing...

Best,

Thomas

**Reply by ghost** [reply]
Hi Thomas,

thank you for the quick answer and sorry for being unclear before.

We're attempting to recreate the quadratic Hill yield criterion as described [here](https://en.wikipedia.org/wiki/Hill_yield_criterion), with the coefficients F,G,H,L,M,N being calculated from our experimental yield stresses.

Before we tackle the `seq` and `iseq` definitions or look at the `@Integrator` at all though: Is it correct that with our parameters being as they are (resulting in a negative Hill coefficient), the Hill tensor as defined by MFront will not work for us "out of the box"? Meaning, this is a general MFront limitation, or are you referring to the Hill yield criterion itself?

Because given this formula for the Hill coefficient H:
$H=\frac{1}{2}\left[{\cfrac {1}{(\sigma _{1}^{y})^{2}}}+{\cfrac {1}{(\sigma _{2}^{y})^{2}}}-{\cfrac {1}{(\sigma _{3}^{y})^{2}}}\right]$

if $\sigma _{1}$ and $\sigma _{2}$ are sufficiently big as in our example, and $\sigma _{3}$ is sufficiently small, there being a negative coefficient is indeed possible. We also couldn't find a definition stating that the coefficients must not be negative. Our values for the yield stresses do satisfy the stability equation (2.13) described by [this paper from Liu (1997)](https://www.sciencedirect.com/science/article/pii/S1359645496003497) providing the limit on the difference of yield stresses for which the yield surface represented by the Quadratic Hill yield criterion remains convex. So from our understanding, the Hill criterion should generally still be applicable even with our parameters resulting in a negative Hill coefficient.

As for the `seq` variable, would you mind elaborating on how exactly the Hill equivalent stress was derived from the yield criterion? In the book of Lemaître et Chaboche, we couldn't find the formulas to compare to the MFront definitions. Since in the docs, it is mentioned that this convention seems to differ from the one described in most other books, this might be relevant to our understanding of the formulas.

Best regards,
Ruben

**Reply by thelfer** [reply]
@aheinri5 Thanks for the reference.

Your formula can be rewritten as follows: 

seq_H / R0 = 1

by taking the square root of the expression given by wikipedia, where seq_H is an equivalent stress and R0 is "an initial yield strength" to be choosen arbitrarly. This means that F,G,H,L,M,N are defined up to a multiplicative coefficient.  

What is implemented here https://tfel.sourceforge.net/orthotropiclinearhardeningplasticity.html just fits in that framework, except that we added a linear hardening rule so that the yield surface is

seq_H = R0 + H * p

Again, it is clear that you can have an equivalent definition by multiplying this equation by a strictly positive scalar alpha (The Hill coefficients would scale with alpha^2).

In many cases, the conventions is given by the fact that L,M,N are assumed equal and are identified to their values for a von Mises equivalent stress.

Anyway, you are right about the fact the constraints on F,G,H, they must only satisfy the following obvious inequalities:

G+H>=0
F+H>=0
F+G>=0

If this is not the case, this indeed means that the Hill model is not able to describe your material.

**Reply by ghost** [reply]
@thelfer thank you very much for the explanation. We figured out why it wasn't working for us - we did not take the orthotropic axis convention into account, so our coefficients were in the wrong order. 

One last question about the axis convention: If we would like to use the `Plate` instead of `Pipe` orthotropic axis convention, how would we set the `ModellingHypotheses`? For example, `{".+"}` as specified in the Hill example would not work because the `AxisymmetricalGeneralisedPlaneStrain` modelling hypothesis is not compatible with `Plate` behaviour, but seperately listing the hypotheses that should be compatible didn't work either. Or is it sufficient to only specify the Hypothesis without specifying the `@OrthotropicBehaviour` as suggested [here](https://tfel.sourceforge.net/MericCailletaudSingleCrystalPlasticity.html), if we are only taking the `Tridimensional` case into account?

Best regards,
Ruben

**Reply by thelfer** [reply]
>  how would we set the ModellingHypotheses? 

You can use the `@ModellingHypotheses` keyword:

```cxx
@ModellingHypotheses {PlaneStress, Tridimensional};
```

> seperately listing the hypotheses that should be compatible didn't work either.

Well, I am pretty sure it does :)
